knowledge-portal/central/central-pipeline-checks.py

```diff
@@ -28,17 +26,16 @@
 
         # Get all lines in the manifest as a list
         lines = f.readlines()
-
-        # Split by '-->', take RHS and strip white spaces
-        rhs_lines = [s.split('-->')[1].strip() for s in lines]
         
-        # Iterate through rhs_lines and get lhs_lines for them with rhs as key and lhs as value
-        lhs_lines = {}
-        for rhs_line in rhs_lines:
-            lhs_lines[rhs_line] = [s.split('-->')[0] for s in lines if s.split('-->')[1].strip() == rhs_line]
-
+        # Check if line contains "-->", if yes, strip the line at "-->" and get source and central paths 
+        paths = {}  # central path as key and source path as value 
+        for line in lines:
+            if "-->" in line:
+                central_path = line.split('-->')[1].strip()
+                paths[central_path] = line.split('-->')[0]    
+        
         # Iterate through RHS and see if we find any matches in changed files
-        for rhs_line in rhs_lines:
-            if rhs_line in changed_files:
-                lhs_line = ''.join(map(str, lhs_lines[rhs_line]))
-                print(f"{repo_name}:{rhs_line}-->{lhs_line}")
+        for central_path in paths.keys():
+            if central_path in changed_files:
```

Comment:
Does this mean that if a new file was added through CKP in a directory that's being synced from a sourceGit, it won't be sent back to sourceGit?
(Ex: sourceGit is syncing a folder `/myfolder/` to `/myCentralGitPrefix/myfolder` at Central Git, and hence it is also showing up at `myCentralGitPrefix/myfolder/` within CKP. Now, the user creates a new file using the CKP editor inside `myCentralGitPrefix/myfolder/` location. Does this new file get sent back to sourceGit?)

If it won't sync back, then it's alright, we should document it as a limitation in the readme

Reply:
I think I have done a boo-boo here, `changed_files` are always going to be filenames but `central_path` can also be a folder path, If that's the case it won't copy anything back to git. I will quickly modify this. 

I guess we want the newly created file on CKP to also sync back to source?

Reply:
If possible and easy to implement, yes 🙂